src/test/java/teammates/logic/core/NotificationsLogicTest.java

```diff
@@ -77,6 +114,65 @@ private void testCreateNotification() throws Exception {
         verifyAbsentInDatabase(n);
     }
 
+    private void testUpdateNotification() throws Exception {
+        ______TS("success: typical case");
+
+        n = dataBundle.notifications.get("notification1");
+
+        NotificationAttributes.UpdateOptions update1 =
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withStyle(NotificationStyle.WARNING)
+                        .withTargetUser(NotificationTargetUser.STUDENT)
+                        .withTitle("The edited title")
+                        .withMessage("The edited message")
+                        .build();
+
+        NotificationAttributes actual = notifLogic.updateNotification(update1);
+        assertEquals(NotificationStyle.WARNING, actual.getStyle());
+        assertEquals(NotificationTargetUser.STUDENT, actual.getTargetUser());
+        assertEquals("The edited title", actual.getTitle());
+        assertEquals("The edited message", actual.getMessage());
+
+        ______TS("failure: invalid build options");
+
+        n = dataBundle.notifications.get("notification1");
+
+        NotificationAttributes.UpdateOptions update2 =
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withTitle("")
+                        .build();
+
+        assertThrows(InvalidParametersException.class,
+                () -> notifLogic.updateNotification(update2));
+
+        ______TS("failure: invalid notification id");
+
+        NotificationAttributes.UpdateOptions update3 =
+                NotificationAttributes.updateOptionsBuilder("invalid-notification-id")
+                        .withTitle("A valid title")
+                        .build();
+
+        assertThrows(EntityDoesNotExistException.class,
+                () -> notifLogic.updateNotification(update3));
+    }
+
+    private void testDeleteNotification() {
+        ______TS("success: delete corresponding notification");
+
+        n = dataBundle.notifications.get("notification1");
+        notifLogic.deleteNotification(n.getNotificationId());
+
+        verifyAbsentInDatabase(n);
+
+        ______TS("failure: invalid id");
```

Comment:
Not sure if this would be considered a failure, since trying to delete a notification with an id not present in the DB would silently fail and is expected behaviour (no errors are thrown)

Reply:
The test is checking that no notification is deleted, i.e. testing for the silent failure

Reply:
Could we add on to the test case descriptor or add a comment? Thinking if the test fails, people might be confused why the expected result is as is, from the current descriptor

Reply:
Comment has been added, thanks for the suggestion!